tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/klass/ClassTest.java

```diff
@@ -64,6 +64,11 @@ public void testTypeRef() {
         BRunUtil.invoke(compileResult, "testTypeRefInClass");
     }
 
+    @Test
+    public void testClassWithDefaultsWithoutInitFunc() {
+        BRunUtil.invoke(compileResult, "testClassWithDefaultsWithoutInitFunc");
```

Comment:
Please add tests with module-level and local object constructors too.

And an actual closure scenario for a local object constructor.